src/pybamm/solvers/base_solver.py

```diff
@@ -1562,103 +1550,55 @@ def report(string):
                 f"to parameters {model.calculate_sensitivities} using jax"
             )
             jacp = func.get_sensitivities()
-        if use_jacobian:
-            report(f"Calculating jacobian for {name} using jax")
-            jac = func.get_jacobian()
-            jac_action = func.get_jacobian_action()
-        else:
-            jac = None
-            jac_action = None
+        jac = func.get_jacobian() if use_jacobian else None
```

Comment:
I think the single if statement is less complex then multiple, plus you are removing the `report`, so I'd disagree with this change